src/modules/todoclass.js

```diff
@@ -0,0 +1,98 @@
+import NewTask from './newtodo.js';
+
+const checkboxes = document.getElementsByClassName('checkbox');
+const todoActivityDiv = document.getElementById('activity-div');
+const removebtn = document.getElementsByClassName('removebtn');
+const desc = document.getElementsByClassName('desc');
+
+class TodoList {
+  constructor() {
+    this.todostorage = JSON.parse(localStorage.getItem('todoList')) || [];
+  }
+
+  addBook = (description) => {
+    const booked = new NewTask(description, false, this.todostorage.length);
+    this.todostorage.push(booked);
+    localStorage.setItem('todoList', JSON.stringify(this.todostorage));
+    return this.todostorage;
+  };
+
+  descChange = (todoId) => {
+    document.getElementById(`${todoId}-${todoId}`).onclick = () => {
+      document.getElementById(`${todoId}-${todoId}`).style.display = 'none';
+      document.getElementById(`${todoId}`).setAttribute('readonly', true);
+      this.todostorage[todoId].description = document.getElementById(
+        `${todoId}`,
+      ).value;
+      localStorage.setItem('todoList', JSON.stringify(this.todostorage));
+    };
+  };
+
+  checkbox = (todoId) => {
+    this.todostorage[todoId].complete = !this.todostorage[todoId].complete;
+    localStorage.setItem('todoList', JSON.stringify(this.todostorage));
+  };
+
+  remove = (todoId) => {
+    const filterFunction = (todo) => todo.index !== parseInt(todoId, 10);
+    const filteredTodos = this.todostorage.filter(filterFunction);
+    this.displayBooks();
+    this.todostorage = filteredTodos;
+    this.RearrangeArray();
+    localStorage.setItem('todoList', JSON.stringify(filteredTodos));
+    return this.todostorage;
+  };
+
+  RearrangeArray = () => {
+    for (let i = 0; i < this.todostorage.length; i += 1) {
+      this.todostorage[i].index = i;
+    }
+  };
```

Comment:
Kindly use the sort method to rearange the arrray as it adds simplicity to your code.